ext/hal/cmsis/DSP/Kconfig

```diff
@@ -0,0 +1,509 @@
+	bool "rfft fast f32 64"
+
+config CMSIS_DSP_TABLES_RFFT_FAST_F32_128
+	bool "rfft fast f32 128"
+
+config CMSIS_DSP_TABLES_RFFT_FAST_F32_256
+	bool "rfft fast f32 256"
+
+config CMSIS_DSP_TABLES_RFFT_FAST_F32_512
+	bool "rfft fast f32 512"
+
+config CMSIS_DSP_TABLES_RFFT_FAST_F32_1024
+	bool "rfft fast f32 1024"
+
+config CMSIS_DSP_TABLES_RFFT_FAST_F32_2048
+	bool "rfft fast f32 2048"
+
+config CMSIS_DSP_TABLES_RFFT_FAST_F32_4096
+	bool "rfft fast f32 4096"
+
+config CMSIS_DSP_TABLES_RFFT_F64_128
+	bool "rfft f64 128"
+
+config CMSIS_DSP_TABLES_RFFT_F64_512
+	bool "rfft f64 512"
+
+config CMSIS_DSP_TABLES_RFFT_F64_2048
+	bool "rfft f64 2048"
+
+config CMSIS_DSP_TABLES_RFFT_F64_8192
+	bool "rfft f64 8192"
+
+config CMSIS_DSP_TABLES_RFFT_F32_128
+	bool "rfft f32 128"
+
+config CMSIS_DSP_TABLES_RFFT_F32_512
+	bool "rfft f32 512"
+
+config CMSIS_DSP_TABLES_RFFT_F32_2048
+	bool "rfft f32 2048"
+
+config CMSIS_DSP_TABLES_RFFT_F32_8192
+	bool "rfft f32 8192"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_32
+	bool "rfft q31 32"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_64
+	bool "rfft q31 64"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_128
+	bool "rfft q31 128"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_256
+	bool "rfft q31 256"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_512
+	bool "rfft q31 512"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_1024
+	bool "rfft q31 1024"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_2048
+	bool "rfft q31 2048"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_4096
+	bool "rfft q31 4096"
+
+config CMSIS_DSP_TABLES_RFFT_Q31_8192
+	bool "rfft q31 8192"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_32
+	bool "rfft q15 32"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_64
+	bool "rfft q15 64"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_128
+	bool "rfft q15 128"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_256
+	bool "rfft q15 256"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_512
+	bool "rfft q15 512"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_1024
+	bool "rfft q15 1024"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_2048
+	bool "rfft q15 2048"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_4096
+	bool "rfft q15 4096"
+
+config CMSIS_DSP_TABLES_RFFT_Q15_8192
+	bool "rfft q15 8192"
+
+config CMSIS_DSP_TABLES_DCT4_F32_128
+	bool "dct4 f32 128"
+
+config CMSIS_DSP_TABLES_DCT4_F32_512
+	bool "dct4 f32 512"
+
+config CMSIS_DSP_TABLES_DCT4_F32_2048
+	bool "dct4 f32 2048"
+
+config CMSIS_DSP_TABLES_DCT4_F32_8192
+	bool "dct4 f32 8192"
+
+config CMSIS_DSP_TABLES_DCT4_Q31_128
+	bool "dct4 q31 128"
+
+config CMSIS_DSP_TABLES_DCT4_Q31_512
+	bool "dct4 q31 512"
+
+config CMSIS_DSP_TABLES_DCT4_Q31_2048
+	bool "dct4 q31 2048"
+
+config CMSIS_DSP_TABLES_DCT4_Q31_8192
+	bool "dct4 q31 8192"
+
+config CMSIS_DSP_TABLES_DCT4_Q15_128
+	bool "dct4 q15 128"
+
+config CMSIS_DSP_TABLES_DCT4_Q15_512
+	bool "dct4 q15 512"
+
+config CMSIS_DSP_TABLES_DCT4_Q15_2048
+	bool "dct4 q15 2048"
+
+config CMSIS_DSP_TABLES_DCT4_Q15_8192
+	bool "dct4 q15 8192"
+
+endif # CMSIS_DSP_TABLES
+
+comment "Instruction Set"
+
+config CMSIS_DSP_NEON
+	bool "Neon Instruction Set"
+	# depends on CPU_CORTEX_A
```

Comment:
Why commented out?

Reply:
`CPU_CORTEX_A` is commented out because it is currently not supported by Zephyr (there is an ongoing AArch64/ARMv8-A PR at the moment). It should be commented out when Cortex-A support is added in the future.

Reply:
Maybe the comment could be tweaked to

```
# Cortex-A is not currently supported. Uncomment once it is.
# depends on CPU_CORTEX_A
```